stm32-modules/heater-shaker/firmware/host_comms_task/freertos_comms_task.cpp

```diff
@@ -27,10 +28,12 @@ struct CommsTaskFreeRTOS {
     USBD_CDC_ItfTypeDef cdc_class_fops;
     USBD_HandleTypeDef usb_handle;
     USBD_CDC_LineCodingTypeDef linecoding;
-
+    UART_HandleTypeDef uart_handle;
     double_buffer::DoubleBuffer<char, CDC_DATA_HS_MAX_PACKET_SIZE * 4> rx_buf;
     double_buffer::DoubleBuffer<char, CDC_DATA_HS_MAX_PACKET_SIZE * 4> tx_buf;
     char *committed_rx_buf_ptr;
+    double_buffer::DoubleBuffer<char, CDC_DATA_HS_MAX_PACKET_SIZE * 4> uart_rx_buf; //make same size as USB buffers?
```

Comment:
it doesn't really have to be, it could be smaller if you want. the USB double buff has to be that size because thats the maximum possible size of a USB packet. the uart needs less but also if we've got the memory available who cares